internal/spec/definition.go

```diff
@@ -85,9 +96,54 @@ func ParseDefinitionFile(content []byte) (*Definition, error) {
 		return nil, err
 	}
 
+	for _, scenario := range specDefinition.Scenarios {
+		for _, nrql := range scenario.Tests.NRQLs {
+			err := validateNRQLTestConfig(nrql)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
 	if specDefinition.CustomTestKey == "" {
 		specDefinition.CustomTestKey = defaultCustomTagKey
 	}
 
 	return specDefinition, nil
 }
+
+func validateNRQLTestConfig(nrqlTest TestNRQL) error {
+	if nrqlTest.Query == "" {
+		return fmt.Errorf("%w: missing query param", ErrInvalidConfig)
+	}
+
+	if nrqlTest.ExpectedResults != nil {
+		// Check expected value config
+		if nrqlTest.ErrorExpected {
+			return fmt.Errorf("%w: expected_results cannot be used with error_expected", ErrInvalidConfig)
+		}
+
+		for i, expectedResult := range nrqlTest.ExpectedResults {
+			err := validateExpectedResult(expectedResult, i)
```

Comment:
Nit: we don't need the index to perform a single `exepctedResult` validation, it could be included in the error message here by wrapping the error.